bigint/src/lib.rs

```diff
@@ -1484,14 +1497,16 @@ impl FromPrimitive for BigUint {
 }
 
 impl From<u64> for BigUint {
-    // `DoubleBigDigit` size dependent
     #[inline]
-    fn from(n: u64) -> Self {
-        match big_digit::from_doublebigdigit(n) {
-            (0, 0) => BigUint::zero(),
-            (0, n0) => BigUint { data: vec![n0] },
-            (n1, n0) => BigUint { data: vec![n0, n1] },
+    fn from(mut n: u64) -> Self {
+        let mut ret: BigUint = Zero::zero();
+
+        while n != 0 {
+            ret.data.push(n as BigDigit);
+            n = (n >> 1) >> (big_digit::BITS - 1);
```

Comment:
I understand you're avoiding shift overflow if `BITS` were 64, but this needs a comment.